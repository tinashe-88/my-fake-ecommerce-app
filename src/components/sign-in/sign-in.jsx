import React, { useState } from 'react'

import FormInput from '../form-input/form-input'
import CustomButton from '../custom-button/custom-button'

import {
  SignInContainer,
  SignInTitle,
  SignInText,
  ButtonsBarContainer
} from './sign-in.styles'

const SignIn = () => {
  const [userCredentials, setUserCredntials] = useState({
    email: '',
    password: ''
  })

  const { email, password } = userCredentials

  const handleSubmit = e => {
    e.preventDefault()

    this.setState({
      email: '',
      password: ''
    })
  }

  const handleChange = e => {
    const { value, name } = e.target

    setUserCredntials({...userCredentials, [name]: value})
  }

  return (
    <SignInContainer>
      <SignInTitle>Already have an account?</SignInTitle>
      <SignInText>Sign in with your credentials</SignInText>

      <form onSubmit={handleSubmit}>
        <FormInput 
          name='email'
          type='email'
          value={email}
          onChange={handleChange}
          label='email'
          required 
        />
        <FormInput 
          name='password'
          type='password'
          value={password}
          onChange={handleChange}
          label='password'
          required
        />
        <ButtonsBarContainer>
          <CustomButton type='submit' value='Submit form'>
            Sign In
          </CustomButton>
        </ButtonsBarContainer>
      </form>
    </SignInContainer>
  )
}

export default SignIn