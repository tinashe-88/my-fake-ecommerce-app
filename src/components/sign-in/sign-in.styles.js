import styled from 'styled-components'

export const SignInContainer = styled.div`
  width: 380px;
  display: flex;
  flex-direction: column;

  @media screen and (max-width: 500px) {
    width: 93%;
  }
`

export const SignInTitle = styled.h2`
  margin: 10px 0;
`

export const SignInText = styled.span`

`

export const ButtonsBarContainer = styled.div`
  display: flex;
  justify-content: space-between;

  @media screen and (max-width: 800px) {
    flex-direction: column;
    &:last-of-type {
      padding-bottom: 10px;
    }
  }
`