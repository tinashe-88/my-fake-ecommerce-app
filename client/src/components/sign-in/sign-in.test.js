import React from 'react'
import { shallow } from 'enzyme'

import { SignIn } from './sign-in'

it('Should render SignIn component', () => {
  expect(shallow(<SignIn />)).toMatchSnapshot()
})