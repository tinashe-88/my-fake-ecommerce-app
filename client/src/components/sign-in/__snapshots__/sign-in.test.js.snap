// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SignIn component 1`] = `
<styled.div>
  <styled.h2>
    I already have an account
  </styled.h2>
  <styled.span>
    Sign in with your email and password
  </styled.span>
  <form
    onSubmit={[Function]}
  >
    <FormInput
      handleChange={[Function]}
      label="email"
      name="email"
      required={true}
      type="email"
      value=""
    />
    <FormInput
      handleChange={[Function]}
      label="password"
      name="password"
      required={true}
      type="password"
      value=""
    />
    <styled.div>
      <CustomButton
        type="submit"
      >
         Sign in 
      </CustomButton>
      <CustomButton
        isGoogleSignIn={true}
        type="button"
      >
        Google Sign In
      </CustomButton>
    </styled.div>
  </form>
</styled.div>
`;
